### ISSUE TITLE
Offcloud debrid support

### ISSUE THREAD

**Issue by Blockr22** [Goldenfreddy0703/Otaku#187]
Hey dev, can u plz add integration for offcloud debrid support sometime in the future ?

**Comment by Goldenfreddy0703** [comment]
Would love to but i don't think we will be adding this anytime soon cause in Otaku, we would need to possibly add a download function to our addon or a place it will check for local files and then we would need to have Otaku analyze the files to scrap off local and all. 

We would basically just need an option that says "Inspect Local Files on Scrape" and then have a option for the user to specify the location for it. 

Btw this is a duplicate to #160

**Comment by Blockr22** [comment]
Hey dev, it’s actually not a duplicate to #160. In that they want is local files scraping. What i want is to have offcloud debrid service support. Its same as real debrid or all debrid. U can check it by going on to offcloud.com

**Comment by Goldenfreddy0703** [comment]
Ohhhh I thought offcloud was like scrapping from local files so your scrapping offline in a way, my bad.

About the offcloud support, I'm not too sure if we could support it, my team may not be interested in adding this cause most kodi addons i seen don't even support offcloud but will discuss this sometime and see what we think. 

Thank you

**Comment by Blockr22** [comment]
Yeah, i see thats its not popular & is only supported by torrentio for Stremio. But i luv otaku for watching anime. It would be great if its added in couple of months. Thnx

**Comment by Gujal00** [comment]
There is no secure oauth2 with pin authentication supported by offcloud. This means we have to ask the user for the email and pasdsword and store in otaku settings. This is a big security risk and users can even blame us for stealing their credentials, So until offcloud support oauth2 with pin authentication like realdebrid, alldebrid, premiumize  and debridlink, it will not be added to Otaku

**Comment by Blockr22** [comment]
Ohh, alright. I understand got it. Thnx

I will ask offcloud customer service, when will that be implemented.

**Comment by Goldenfreddy0703** [comment]
Hey so im going to go ahead and close this for now but once you get in touch with there customer service and all, we will open this back up and look into this if they reach out. 

Thank you

**Comment by Blockr22** [comment]
Kk, thnx. Np
I just reached to their customer & will let u know what do they say

**Comment by Blockr22** [comment]
Hi, so since my last comment few more community Addons support Offcloud debrid that are Mediafusion for Stremio and POV for Kodi in addition to Torrentio. Annatar and Jackett also have people asking for Offcloud debrid integration since there are people like me who bought lifetime sub for Offcloud as a backup in addition to real debrid. 
Also did put a support ticket to Offcloud for support OAuth2 like that in real debrid or alldebrid and this was there response more than a month ago

'Thank you for your inquiry. We truly appreciate your feedback and the effort to suggest features that could enhance the security and usability of Offcloud. We'll take the suggestion into account.

Please stay tuned for updates on our features and offerings. If you have any more ideas or concerns, do not hesitate to reach out again.

Best regards,
The Offcloud Support Team'

I have contacted them again today for any ETA but i don't know when will this be implemented by them. I hope u can implement Offcloud support like others did with some of their addons. Thnx

**Comment by Goldenfreddy0703** [comment]
Ok so you told me the POV addon supports Kodi and i do see offcloud and also easynews and from looking in its code, it kinda looks easy to understand. Maybe we can look into this on a later date. 
Thank you

**Comment by Blockr22** [comment]
That would be great if u can implement it in the near future. Thnku so much
Here are the links which will help i think

https://github.com/Offcloud/offcloud-api
https://offcloud.com/

**Comment by Blockr22** [comment]
Hey dev, is this being worked on yet and can u plz open this issue. Maybe someone can help when its visible to others too. thnx

**Comment by Goldenfreddy0703** [comment]
Hey so it's not being worked on at the moment cause some of our contributors have been busy this month but hopefully we will get back to this around April. 

I will go ahead and leave this issue open sense you got in contact with them and that there is a kodi addon having this implemented.

**Comment by Blockr22** [comment]
Thnx for the update & opening this feature request.

**Comment by Blockr22** [comment]
Hey dev, any update with my request ?

**Comment by Goldenfreddy0703** [comment]
Hey sorry i did not get back to you, unfortunately there is not, we need an account but we are lazy to pay for one lol. Maybe in the future we will see if we can assign @joezito129 to this sense he worked on alldebrid cloud for us.

**Comment by Blockr22** [comment]
Ohh, kk. Thnx
I hope it’s supported soon.
Let me know if sharing my account temporarily helps.
